test/e2e/serial/tests/sched_removal.go

```diff
@@ -137,10 +139,29 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources scheduler restar
 		It("[case:1][test_id:48069][tier2] should schedule any pending workloads submitted while the scheduler was unavailable", func() {
 			var err error
 
+			dps, err := getDPOwnedBy(fxt.Client, nroSchedObj.ObjectMeta)
+			Expect(err).ToNot(HaveOccurred())
+
 			By(fmt.Sprintf("deleting the NRO Scheduler object: %s", nroSchedObj.Name))
 			err = fxt.Client.Delete(context.TODO(), nroSchedObj)
 			Expect(err).ToNot(HaveOccurred())
 
+			// make sure scheduler deployment is gone
+			Eventually(func() bool {
+				for _, dp := range dps {
+					if err := fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(dp), dp); err != nil {
+						if apierrors.IsNotFound(err) {
+							return true
```

Comment:
it seems to me this means we bail out as soon as the first deployment is gone - but we should actually wait for the _last_ to be gone (meaning all of them are gone), right?

Reply:
We should only have a single deployment for the scheduler.

We can actually assert there is only one and failed the test if it's different.

Reply:
I'd prefer this way. `getDPOwnedBy` should either return 1 Deployment (success) or nil (on error).